harness/determined/_local_execution.py

```diff
@@ -0,0 +1,115 @@
+    configuration with each entire subconfig specified by a user.
+
+    The defaults and merging logic is not guaranteed to match the logic used by
+    the Determined master. This function also does not do experiment
+    configuration validation, which the Determined master does.
+    """
+
+    input_config = input_config.copy() if input_config else {}
+    config_keys_to_ignore = {
+        "bind_mounts",
+        "checkpoint_storage",
+        "environment",
+        "resources",
+        "optimizations",
+    }
+    for key in config_keys_to_ignore:
+        if key in input_config:
+            logging.info(
+                f"'{key}' configuration key is not supported by local test mode and will be ignored"
+            )
+            del input_config[key]
+
+    return {**constants.DEFAULT_EXP_CFG, **input_config}
+
+
+def _make_local_execution_env(
+    config: Optional[Dict[str, Any]], hparams: Optional[Dict[str, Any]] = None,
+) -> Tuple[det.EnvContext, det.RendezvousInfo, horovod.HorovodContext]:
+    config = det.ExperimentConfig(_make_local_execution_exp_config(config))
+    hparams = hparams or api.generate_random_hparam_values(config.get("hyperparameters", {}))
+    use_gpu, container_gpus, slot_ids = _get_gpus()
+    local_rendezvous_ports = (
+        f"{constants.LOCAL_RENDEZVOUS_PORT},{constants.LOCAL_RENDEZVOUS_PORT+1}"
+    )
+
+    env = det.EnvContext(
+        master_addr="",
+        master_port=0,
+        container_id="",
+        experiment_config=config,
+        hparams=hparams,
+        initial_workload=workload.train_workload(1, 1, 1, config.batches_per_step()),
+        latest_checkpoint=None,
+        use_gpu=use_gpu,
+        container_gpus=container_gpus,
+        slot_ids=slot_ids,
+        debug=config.debug_enabled(),
+        workload_manager_type="",
+        det_rendezvous_ports=local_rendezvous_ports,
+        det_trial_runner_network_interface=constants.AUTO_DETECT_TRIAL_RUNNER_NETWORK_INTERFACE,
+        det_trial_id="",
+        det_experiment_id="",
+        det_cluster_id="",
+        trial_seed=config.experiment_seed(),
+    )
+    rendezvous_ports = env.rendezvous_ports()
+    rendezvous_info = det.RendezvousInfo(
+        addrs=[f"0.0.0.0:{rendezvous_ports[0]}"], addrs2=[f"0.0.0.0:{rendezvous_ports[1]}"], rank=0
+    )
+    hvd_config = horovod.HorovodContext.from_configs(
+        env.experiment_config, rendezvous_info, env.hparams
+    )
+
+    return env, rendezvous_info, hvd_config
+
+
+@contextlib.contextmanager
+def _local_execution_manager(context_dir: pathlib.Path) -> Iterator:
+    """
+    A context manager used for local execution to mimic the environment of trial
+    container execution.
+
+    It does the following things:
+    1. Set the current working directory to be the context directory.
+    2. Add the current working directory to importing paths.
+    """
+    # TODO(DET-2719): Add context dir to TrainContext and remove this function.
```

Comment:
Can we remove this TODO?  I looked at the ticket and the associated PR and I'm not convinced that we should do it at all.